From d2e69fe9c6765ecdb6c6edce36fa1bc05ac65bca Mon Sep 17 00:00:00 2001
From: Razvan Stefanescu <razvan.stefanescu@nxp.com>
Date: Tue, 3 Apr 2018 13:53:00 +0300
Subject: [PATCH 468/767] staging: dpaa2-evb: Use MC portal in atomic context

Avoid triggering stack trace when retrieving interface counters via
ifconfig by allocating MC portal with atomic I/O enabled.

Signed-off-by: Razvan Stefanescu <razvan.stefanescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/evb/evb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/evb/evb.c b/drivers/staging/fsl-dpaa2/evb/evb.c
index 3f0782f5a7fa..219c4758d316 100644
--- a/drivers/staging/fsl-dpaa2/evb/evb.c
+++ b/drivers/staging/fsl-dpaa2/evb/evb.c
@@ -1202,7 +1202,8 @@ static int evb_probe(struct fsl_mc_device *evb_dev)
 
 	priv = netdev_priv(netdev);
 
-	err = fsl_mc_portal_allocate(evb_dev, 0, &priv->mc_io);
+	err = fsl_mc_portal_allocate(evb_dev, FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
+				     &priv->mc_io);
 	if (unlikely(err)) {
 		dev_err(dev, "fsl_mc_portal_allocate err %d\n", err);
 		goto err_free_netdev;
-- 
2.17.0

